diff -ruN ./base/g.py ../hplipMod/base/g.py
--- ./base/g.py	2019-06-26 18:37:10.000000000 +0530
+++ ../hplipMod/base/g.py	2019-07-19 03:45:00.832825111 +0530
@@ -182,16 +182,24 @@
         
 class SysConfig(ConfigBase):
     def __init__(self):
-        ConfigBase.__init__(self, '/etc/hp/hplip.conf')
+        snap = os.getenv("SNAP")
+        if not snap:
+            snap=""
+
+        ConfigBase.__init__(self, snap+'/etc/hp/hplip.conf')
 
 
 class State(ConfigBase):
     def __init__(self):
-        if not os.path.exists('/var/lib/hp/') and os.geteuid() == 0:
-            os.makedirs('/var/lib/hp/')
-            cmd = 'chmod 755 /var/lib/hp/'
+        snap = os.getenv("SNAP")
+        if not snap:
+            snap=""
+
+        if not os.path.exists(snap+'/var/lib/hp/') and os.geteuid() == 0:
+            os.makedirs(snap+'/var/lib/hp/')
+            cmd = 'chmod 755 %s/var/lib/hp/'%snap
             os_utils.execute(cmd)
-        ConfigBase.__init__(self, '/var/lib/hp/hplip.state')
+        ConfigBase.__init__(self, snap+'/var/lib/hp/hplip.state')
 
 
 class UserConfig(ConfigBase):
diff -ruN ./fax/backend/hpfax.py ../hplipMod/fax/backend/hpfax.py
--- ./fax/backend/hpfax.py	2019-06-26 18:37:00.000000000 +0530
+++ ../hplipMod/fax/backend/hpfax.py	2019-07-19 04:42:32.368989693 +0530
@@ -49,7 +49,10 @@
 
 job_id = 0
 pid = os.getpid()
-config_file = '/etc/hp/hplip.conf'
+snap = os.getenv("SNAP")
+if snap==None:
+    snap = ""
+config_file = snap+'/etc/hp/hplip.conf'
 home_dir = ''
 
 
@@ -64,6 +67,7 @@
 
     try:
         home_dir = config.get('dirs', 'home')
+        home_dir = snap+home_dir
     except:
         bug("Error setting home directory: home= under [dirs] not found.")
         sys.exit(1)
@@ -251,6 +255,9 @@
     else:
         tmp_dir = "/tmp"
 
+    if os.getenv("SNAP_COMMON"):
+        tmp_dir = os.getenv("SNAP_COMMON")
+
     pipe_name = os.path.join(tmp_dir, "hp_fax-pipe-%d" % job_id)
 
     # Create the named pipe. Make sure it exists before sending
